flytekit/common/tasks/notebook.py

```diff
@@ -0,0 +1,542 @@
+                    memory_request
+                )
+            )
+
+        limits = []
+        if storage_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.STORAGE,
+                    storage_limit
+                )
+            )
+        if cpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.CPU,
+                    cpu_limit
+                )
+            )
+        if gpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.GPU,
+                    gpu_limit
+                )
+            )
+        if memory_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.MEMORY,
+                    memory_limit
+                )
+            )
+
+        return  _sdk_runnable.SdkRunnableContainer(
+            command=[],
+            args=[],
+            resources=_task_models.Resources(limits=limits, requests=requests),
+            env=environment,
+            config={}
+        )
+
+
+
+
+def spark_notebook(
+        cache_version='',
+        retries=0,
+        deprecated='',
+        cache=False,
+        timeout=None,
+        notebook_path=None,
+        inputs=None,
+        outputs=None,
+        environment=None,
+):
+    """
+    Decorator to create a Notebook spark task. This task will connect to a Spark cluster, configure the environment,
+    and then execute the code within the notebook_path as the Spark driver program.
+    """
+    return SdkNotebookSparkTask(
+            discovery_version=cache_version,
+            retries=retries,
+            deprecated=deprecated,
+            discoverable=cache,
+            timeout=timeout or _datetime.timedelta(seconds=0),
+            notebook_path=notebook_path,
+            inputs=inputs,
+            outputs=outputs,
+            environment=environment or {},
+        )
+
+
+class SdkNotebookSparkTask(SdkNotebookTask):
+
+    """
+    This class includes the additional logic for building a task that executes Spark Notebooks.
+
+    """
+
+    def __init__(
+            self,
+            notebook_path,
+            discovery_version='',
+            retries=0,
+            deprecated='',
+            discoverable=False,
+            timeout=_datetime.timedelta(seconds=0),
+            inputs=None,
+            outputs=None,
+            environment=None,
+    ):
+
+        spark_exec_path = _os.path.abspath(spark_executor.__file__)
+        if spark_exec_path.endswith('.pyc'):
+            spark_exec_path = spark_exec_path[:-1]
+        self._notebook_path = notebook_path
+
+        # Parse Spark_conf from notebook
+        with open(notebook_path) as json_file:
```

Comment:
we should try and patch the notebook path by first checking if it is relative or absolute
https://stackoverflow.com/questions/3320406/how-to-check-if-a-path-is-absolute-path-or-relative-path-in-cross-platform-way-w

Reply:
done